Dockerfile

```diff
@@ -44,35 +36,51 @@ RUN set -xe; \
     \
     rm --recursive --force /var/lib/apt/lists/*;
 
+# Copy code to container
+COPY --chown=thunder:thunder www ${INSTALLATION_DIRECTORY}
+
+# Unfortunately this doesn't quite work as expected.
+# ENV APACHE_RUN_USER=thunder
+
 # Copy run scripts
-COPY scripts/docker/thunder-php-install /usr/local/bin/
-COPY scripts/docker/thunder-php-test /usr/local/bin/
-COPY scripts/docker/thunder-php-run /usr/local/bin/
+COPY scripts/docker/* /usr/local/bin/
 
-# Set executable
+# Set the scripts to be executable. Create a link to thunder-php-test as this is
+# name expected by the runner.
+ENV INSTALLATION_DIRECTORY=${INSTALLATION_DIRECTORY}
 RUN set -xe; \
     \
-    chmod +x /usr/local/bin/thunder-php-install; \
+    echo -e "\nexport PATH=\"\$PATH:${INSTALLATION_DIRECTORY}/vendor/bin:${INSTALLATION_DIRECTORY}/bin\"\n" >> /home/thunder/.profile; \
     \
-    chmod +x /usr/local/bin/thunder-php-test; \
+    chmod +x /usr/local/bin/drupal-php-install; \
     \
-    chmod +x /usr/local/bin/thunder-php-run;
-
-# Copy pre-build Thunder project to container
-COPY --chown=thunder:thunder www /home/thunder/www
-
-# Install Elastic APM
-RUN set -xe; \
+    chmod +x /usr/local/bin/drupal-php-test; \
+    \
+    chmod +x /usr/local/bin/drupal-php-run; \
+    \
+    chmod +x /usr/local/bin/set-docroot; \
+    \
+    chmod +x /usr/local/bin/install-elastic-apm; \
     \
-    su - thunder --command="cd /home/thunder/www/docroot/core; yarn add elastic-apm-node --dev"; \
+    ln -sfn /usr/local/bin/drupal-php-test /usr/local/bin/thunder-php-test; \
     \
-    su - thunder --command="yarn cache clean"; \
+    set-docroot; \
     \
-    su - thunder --command="composer global require drush/drush"; \
+    cat /home/thunder/.profile;
+
+# Install Elastic APM - this is run as a shell script so we can install in the
+# doc root.
+RUN set -xe; \
     \
-    echo -e "\nexport PATH=\"\$PATH:/home/thunder/.composer/vendor/bin\"\n" >> /home/thunder/.profile; \
+    su - thunder --command="install-elastic-apm";
+
+# Provision core tests
+# @todo move this to one of our modules only work with core checkout atm.
+RUN set -xe; \
```

Comment:
Each RUN statement increases the docker image size, we could combine the last three RUN  statements.